### PR TITLE
docs: update docs for go files

### PR DESCRIPTION
**Pull request by johson-ll**
#### Type of change

- Bug fix
- New feature
- Improvement (improvement to code, performance, etc)
- Test update
- Documentation update

### CODE REVIEW

**Review by bestbeforetoday — COMMENTED** [review]
Many of the comments within the code simply state exactly the same thing as the line of code they are commenting. I think this actually detracts from the readability and maintainability of the code, and should should be avoided.

For example:

```go
// Unmarshal the payload from the envelope.
payload, err := UnmarshalPayload(envelope.Payload)
```

The code itself is already perfectly clear.

**Comment by bestbeforetoday** [comment]
See the [contributing guide](https://hyperledger-fabric.readthedocs.io/en/latest/CONTRIBUTING.html#legal-stuff) for why the PR build has failed the DCO check.

File: protoutil/blockutils.go [hunk 1]
```diff
@@ -242,18 +261,31 @@ type policy interface { // copied from common.policies to avoid circular import.
 	EvaluateSignedData(signatureSet []*SignedData) error
 }
 
+
+// BlockSignatureVerifier returns a BlockVerifierFunc that validates the signatures of a block based on the provided policy and consenter set.
+// It checks the metadata of the block to retrieve and verify the signatures against the given policy.
```
Comment (denyeart) [inline protoutil/blockutils.go:266]: For functions that are missing a function comment, I agree they should be added. But I would keep them to one or two sentences (like above), and avoid the full listing of parameters (like below).

**Review by denyeart — COMMENTED** [review]
I agree with the comment from @bestbeforetoday .

Thank you for adding the leading function comments, those should indeed exist for all exported functions and it is good to add them.

For the comments within the functions, most of them here are overkill and will bloat the code. We typically only add comments within the functions if the code intent is not clear (some of the updates here help, but most of them are unnecessary).

Please update the PR to reduce the number of comments.

**Comment by denyeart** [comment]
More DCO guidance can be found by clicking the link in the failed check:
https://github.com/hyperledger/fabric/pull/5058/checks?check_run_id=33591103081

**Pull request closed without merging**